coderd/healthcheck/derphealth/derp.go

```diff
@@ -209,15 +209,14 @@ func (r *RegionReport) Run(ctx context.Context) {
 
 	sortNodeReports(r.NodeReports)
 
-	// Coder allows for 1 unhealthy node in the region, unless there is only 1 node.
 	if len(r.Region.Nodes) == 1 {
-		r.Healthy = healthyNodes == len(r.Region.Nodes)
+		r.Healthy = !(r.NodeReports[0].Severity == health.SeverityError)
```

Comment:
Ensure `len(r.NodeReports) > 0`. I know that it's unlikely but I'd rather not have to worry about a NPE here.

Reply:
Done, but a added a bit different condition.